soda_machine.rb

```diff
@@ -7,12 +9,26 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      return soda if soda.brand == soda_brand
+    end
+      return nil
   end
 
   def sell(soda_brand)
+    if find_soda(soda_brand) != nil
+      @sodas.each do |soda|
+        if soda.brand == soda_brand
+          @cash += soda.price
+          @sodas.delete(soda)
+          return nil
+        end
+      end
+    end
```

Comment:
You're duplicating effort -- first you're finding it and then throwing it away (in the intial `if`), then finding it again before adding the price and deleting it.  Instead, cache the result of `find_soda` OR get rid of that call and add a return nil at the end.